drivers/timer/stm32_lptim_timer.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2018 Foundries.io Ltd
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <drivers/timer/system_timer.h>
+#include <sys_clock.h>
+
+#include <spinlock.h>
+
+/*
+ * This is just a getting started point.
+ *
+ * Assumptions and limitations:
+ *
+ * - system clock based on an LPTIM1 instance, clocked by LSI or LSE
+ */
+#define LPTIM_CLOCK LSI_VALUE
+
+/* LPTIM counter for a 4s timebase */
+#if (LPTIM_CLOCK == LSI_VALUE)
```

Comment:
Please use Kconfig for configuration

Reply:
get the  LPTIM_CLOCK value through KConfig